packages/server-wallet/src/models/channel.ts

```diff
@@ -173,6 +177,67 @@ export class Channel extends Model implements RequiredColumns {
     });
   }
 
+  public static directFundingStatus(
+    supported: SignedStateVarsWithHash | undefined,
+    fundingFn: (address: Address) => ChannelStateFunding,
+    myParticipant: Participant,
+    fundingStrategy: FundingStrategy
+  ): FundingStatus {
+    if (fundingStrategy !== 'Direct') {
+      return 'Uncategorized';
+    }
+
+    const outcome = supported?.outcome;
+    if (!supported || !outcome) {
+      return 'Uncategorized';
+    }
+
+    const {allocationItems, assetHolderAddress} = checkThat(outcome, isSimpleAllocation);
+
+    // Collapse all allocation items with my destination into one
+    const myAllocation = allocationItems
+      .filter(ai => ai.destination === myParticipant.destination)
+      .reduce(
+        (soFar, currentAi) => ({
+          ...soFar,
+          amount: BN.add(soFar.amount, currentAi.amount),
+        }),
+        {
+          destination: myParticipant.destination,
+          amount: BN.from(0),
+        }
+      );
+
+    const funding = fundingFn(assetHolderAddress);
```

Comment:
I am no longer sure the `fundingFn` does what it was meant to do. I don't even remember why it was a function in the first place 😆

Reply:
`funding` should probably be a dictionary. https://github.com/statechannels/statechannels/issues/2969